src/Hpack.hs

```diff
@@ -81,7 +81,10 @@ data Options = Options {
 , optionsToStdout :: Bool
 }
 
-data GenerateHashStrategy = ForceHash | ForceNoHash | PreferHash | PreferNoHash
+data GenerateHashStrategy
+  = ForceHash     -- ^ Option @--hash@ given.
+  | ForceNoHash   -- ^ Option @--no-hash given.
+  | PreferNoHash  -- ^ None of these option given, default behavior.
```

Comment:
I think you are right that as far as the `hpack` executable is concerned this is indeed dead code.  However, it is exposed through the API, so as I understand it a user of the API could still use it to force pre-0.34 behavior.

Reply:
Ok, so maybe phase this out slowly, or deprecate it in the API first.